File: src/main/java/org/example/coin_laundry_app_backend/order/domain/entity/OrderShippingAddress.java
package org.example.coin_laundry_app_backend.order.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.example.coin_laundry_app_backend.common.entity.AbstractBaseEntity;
import org.example.coin_laundry_app_backend.user.domain.enums.EntranceType;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Getter
@Table("order_shipping_addresses")
@AllArgsConstructor
public class OrderShippingAddress extends AbstractBaseEntity {

    @Id
    private Long id;
    private Long orderId;
    private String addressLabel;
    private String recipientName;
    private String recipientPhone;
    private String baseAddress;
    private String detailAddress;
    private Double latitude;
    private Double longitude;
    private String deliveryNotes;
    private String entranceType;
    private String entranceDetail;
}

File: carry-order/src/main/kotlin/com/carry/order/domain/vo/OrderShippingAddress.kt
package com.carry.order.domain.vo

import com.carry.common.exception.requireInput

data class OrderShippingAddress(
    val roadAddress: String,
    val detailAddress: String,
    val zipCode: String?,
    val latitude: Double,
    val longitude: Double,
    val recipientName: String,
    val recipientPhone: String,
    val entranceInfo: String?,
    val areaCode: String,
) {
    init {
        requireInput(roadAddress.isNotBlank()) { "도로명 주소는 필수입니다" }
        requireInput(recipientName.isNotBlank()) { "수령인 이름은 필수입니다" }
        requireInput(recipientPhone.isNotBlank()) { "수령인 전화번호는 필수입니다" }
        requireInput(areaCode.isNotBlank()) { "지역 코드는 필수입니다" }
    }
}
